### PR TITLE
Questionnaires that have the confirmationDialog config are not extracted | 0.2.3 - Release

### LINKED ISSUE #2489: Questionnaires that have the `confirmationDialog` config are not extracted.
**Describe the bug**
- When launching a questionnaire you can define the message to show on the confirmation dialog that appears after clicking save. 
- This means the Questionnaire extraction only after clicking the Positive button on the dialog. 
- Questionnaires closed this way are not extracted because they do not go through the `QuestionnaireViewModel.extractAndSaveResources` method. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Update any questionnaire launch config and add the `confirmationDialog` configs. 
    - I used the eCBIS remove family [configs](https://github.com/opensrp/fhir-resources/blob/7f6bceb3bd62064899cda24ecbe3becaabcc3fa2/ecbis_liberia/app_configurations/ecbis_cha_preview/profiles/household_profile_config.json#L1364-L1368). 
3. Fill up the questionnaire and submit it. 
4. Sync to the server or check your local SQLite DB for the newly generated resources.  This includes the QuestionnaireResponse. 

**Expected behaviour**
- [ ] All questionnaires should be extracted except when they are draft.

### PR DESCRIPTION
**Pull request by dubdabasoduba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2489  or Closes #2489 

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by dubdabasoduba** [comment]
@ndegwamartin Please share the correct branch to use for the latest release. This PR should target that branch

**Comment by ndegwamartin** [comment]
> #2489

The release was from the `main` branch. We always strive to make releases from this branch.

**Comment by ndegwamartin** [comment]
> @ndegwamartin Please share the correct branch to use for the latest release. This PR should target that branch


The release was from the `main` branch. We always strive to make releases from this branch, so this PR is correct.

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt [hunk 1]
```diff
@@ -510,6 +516,14 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
...
           )
         }
+
+        questionnaireViewModel.extractAndSaveResources(
+          context = this,
+          questionnaire = questionnaire,
+          questionnaireResponse = questionnaireResponse,
+          questionnaireConfig = questionnaireConfig
+        )
+
```
Comment (ndegwamartin) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:526]: Do we need to set this status to **COMPLETED** as well like in [this line](https://github.com/opensrp/fhircore/blob/2005e8abf11f704a928441b72cc5193e8342ffda/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt#L471)?
`questionnaireResponse.status = QuestionnaireResponse.QuestionnaireResponseStatus.COMPLETED`

If so we can refactor [this block](https://github.com/opensrp/fhircore/blob/2005e8abf11f704a928441b72cc5193e8342ffda/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt#L471) into a class method that can be re-used in both places.

Comment (dubdabasoduba) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:526]: Nice catch we probably should.

**Comment by dubdabasoduba** [comment]
> > @ndegwamartin Please share the correct branch to use for the latest release. This PR should target that branch
> 
> The release was from the `main` branch. We always strive to make releases from this branch, so this PR is correct.

Makes sense. At what point are we going to have a branch for the latest release? We might want to freeze the code on that.

**Comment by ndegwamartin** [comment]
> > > @ndegwamartin Please share the correct branch to use for the latest release. This PR should target that branch
> > 
> > 
> > The release was from the `main` branch. We always strive to make releases from this branch, so this PR is correct.
> 
> Makes sense. At what point are we going to have a branch for the latest release? We might want to freeze the code on that.

QA has not started yet so if we can get this in we can build a release with this patch and then freeze that as 0.2.4 RC 1

**Comment by ndegwamartin** [comment]
> > > @ndegwamartin Please share the correct branch to use for the latest release. This PR should target that branch
> > 
> > 
> > The release was from the `main` branch. We always strive to make releases from this branch, so this PR is correct.
> 
> Makes sense. At what point are we going to have a branch for the latest release? We might want to freeze the code on that.

AFAIK QA has not started on the 0.2.3 release yet so if we can get this merged in we can build a release with the patch and then freeze the codebase as `0.2.4 RC 1` cc @Mstjamush

**Comment by ndegwamartin** [comment]
> > > @ndegwamartin Please share the correct branch to use for the latest release. This PR should target that branch
> > 
> > 
> > The release was from the `main` branch. We always strive to make releases from this branch, so this PR is correct.
> 
> Makes sense. At what point are we going to have a branch for the latest release? We might want to freeze the code on that.

AFAIK QA has not started on the 0.2.3 release yet so if we can get this merged and build a release with the patch, then freeze the codebase as `0.2.4 RC 1` cc @Mstjamush

**Review by pld — APPROVED** [review]

**Pull request merged**